Libraries/LibJS/AST.cpp

```diff
@@ -177,6 +247,38 @@ void BinaryExpression::dump(int indent) const
     case BinaryOp::TypedEquals:
         op_string = "===";
         break;
+    case BinaryOp::TypedInequals:
+        op_string = "!==";
+        break;
+    case BinaryOp::Greater:
+        op_string = ">";
+        break;
+    case BinaryOp::Smaller:
+        op_string = "<";
+        break;
+    case BinaryOp::BitAnd:
+        op_string = "&";
+        break;
+    case BinaryOp::BitOr:
+        op_string = "|";
+        break;
+    case BinaryOp::BitXor:
+        op_string = "^";
+        break;
+    case BinaryOp::BitLeftShift:
+        op_string = "<<";
+        break;
+    case BinaryOp::BitRightShift:
+        op_string = ">>";
+        break;
+    case BinaryOp::BitNot:
+        op_string = "~";
```

Comment:
BitNot is a unary operator and should have its own class.

Reply:
You're definitely right, done, I'll also take the liberty of moving the `not` logical operator to the new unary class.